dialogue-core/src/main/java/com/palantir/dialogue/core/Config.java

```diff
@@ -40,11 +51,56 @@
     default ClientConfiguration clientConf() {
         return ClientConfiguration.builder()
                 .from(rawConfig())
+                .uris(rawConfig().uris().stream().map(Config::stripMeshPrefix).collect(Collectors.toList()))
                 .taggedMetricRegistry(
                         VersionedTaggedMetricRegistry.create(rawConfig().taggedMetricRegistry()))
                 .build();
     }
 
+    enum MeshMode {
+        DEFAULT_NO_MESH,
+        USE_EXTERNAL_MESH
+    }
+
+    @Value.Derived
+    default MeshMode mesh() {
+        long meshUris = rawConfig().uris().stream()
+                .filter(s -> s.startsWith(MESH_PREFIX))
+                .count();
+        long normalUris = rawConfig().uris().stream()
+                .filter(s -> !s.startsWith(MESH_PREFIX))
+                .count();
+
+        if (meshUris > 1) {
+            log.warn(
+                    "Not expecting multiple 'mesh-' prefixed uris - please double-check the uris",
+                    SafeArg.of("meshUris", meshUris),
+                    SafeArg.of("normalUris", normalUris),
+                    SafeArg.of("channel", channelName()));
```

Comment:
I think we discussed throwing in these cases rather than allowing unexpected behavior. I'd much rather start strict and back away than attempt the inverse.